src/Tools/ExternalAccess/FSharp/Internal/SignatureHelp/FSharpSignatureHelpProvider.cs

```diff
@@ -30,29 +30,36 @@ public async Task<SignatureHelpItems> GetItemsAsync(Document document, int posit
             var mappedTriggerInfo = new FSharpSignatureHelpTriggerInfo(mappedTriggerReason, triggerInfo.TriggerCharacter);
             var mappedSignatureHelpItems = await _provider.GetItemsAsync(document, position, mappedTriggerInfo, cancellationToken).ConfigureAwait(false);
 
-            return new SignatureHelpItems(
-                mappedSignatureHelpItems.Items.Select(x =>
-                    new SignatureHelpItem(
-                        x.IsVariadic,
-                        x.DocumentationFactory,
-                        x.PrefixDisplayParts,
-                        x.SeparatorDisplayParts,
-                        x.SuffixDisplayParts,
-                        x.Parameters.Select(y =>
-                            new SignatureHelpParameter(
-                                y.Name,
-                                y.IsOptional,
-                                y.DocumentationFactory,
-                                y.DisplayParts,
-                                y.PrefixDisplayParts,
-                                y.SuffixDisplayParts,
-                                y.SelectedDisplayParts)).ToList(),
-                        x.DescriptionParts)).ToList(),
-                mappedSignatureHelpItems.ApplicableSpan,
-                mappedSignatureHelpItems.ArgumentIndex,
-                mappedSignatureHelpItems.ArgumentCount,
-                mappedSignatureHelpItems.ArgumentName,
-                mappedSignatureHelpItems.SelectedItemIndex);
+            if (mappedSignatureHelpItems != null)
+            {
+                return new SignatureHelpItems(
+                    mappedSignatureHelpItems.Items?.Select(x =>
```

Comment:
[This](http://source.roslyn.io/#Microsoft.CodeAnalysis.Features/SignatureHelp/SignatureHelpItems.cs,63) will throw if it's null and also if it's empty.  The whole function should return `null` instead.

Reply:
Sure, but at least it will throw inside `SignatureHelpItems`. It's going to throw no matter what if that is null. This just adds another guard check.